starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,324 @@
+	return (y + map_size[1]) % map_size[1];
+}
+
+// Calculate the distance betwen two sets of x and y values
+inline unsigned char calc_dist(unsigned char ax, unsigned char ay, unsigned char bx, unsigned char by){
+	unsigned char dx, dy;
+
+	if(ax > bx)
+		dx = ax - bx;
+	else
+		dx = bx - ax;
+
+	if(ay > by)
+		dy = ay - by;
+	else
+		dy = by - ay;
+
+	return (dx % mhw) + (dy % mhh);
+}
+
+// 0 if not occupied, 1 if occupied
+inline unsigned char tile_occupied(unsigned char x, unsigned char y){
+	return map[x][y].occupied != (unsigned short)-1;
+}
+
+// Now for the commands to send to the game engine
+inline void move_still(unsigned short ship_id){
+	printf("m %hu o ", ship_id);
+}
+
+inline void move_north(unsigned short ship_id){
+	printf("m %hu n ", ship_id);
+}
+
+inline void move_south(unsigned short ship_id){
+	printf("m %hu s ", ship_id);
+}
+
+inline void move_east(unsigned short ship_id){
+	printf("m %hu e ", ship_id);
+}
+
+inline void move_west(unsigned short ship_id){
+	printf("m %hu w ", ship_id);
+}
+
+
+inline void convert_dropoff(unsigned short ship_id){
+	printf("c %hu ", ship_id);
+}
+
+inline void spawn_ship(){
+	printf("g ");
+}
+
+
+
+// Now that all of the variables have been declared, we can write some basic "smart" logic
+void random_move(unsigned short ship_id){
+	unsigned char decision = rand() % 100;
+
+	// 10% chance each to move north/east/south/west (This is extra smart)
+	if(decision <= 10){
+		move_north(ship_id);
+	}else if(decision <= 20){
+		move_east(ship_id);
+	}else if(decision <= 30){
+		move_south(ship_id);
+	}else if(decision <= 40){
+		move_west(ship_id);
+	}else{
+		// And a 60% chance to stay still (This is also extra smart)
+		// We could alternatively just not give the ship a move this turn
+		move_still(ship_id);
+	}
+}
+
+
+// There's something that random_move is forgetting... Let's avoid collisions!
+void naive_navigate(unsigned short ship_id){
+	unsigned char decision = rand() % 100, x = ships[ship_id].x, y = ships[ship_id].y;
+
+	// North is y - 1
+	// South is y + 1
+	// West is x - 1
+	// East is x + 1
+
+	// 10% chance each to move north/east/south/west (This is even smarter than before!)
+	if(decision <= 10 && tile_occupied(x, norm_y(y - 1)) == 0){
+		move_north(ship_id);
+	}else if(decision <= 20 && tile_occupied(norm_x(x + 1), y) == 0){
+		move_east(ship_id);
+	}else if(decision <= 30 && tile_occupied(x, norm_y(y + 1)) == 0){
+		move_south(ship_id);
+	}else if(decision <= 40 && tile_occupied(norm_x(x - 1), y) == 0){
+		move_west(ship_id);
+	}else{
+		// And a 60% chance to stay still (This is also smarter than before!)
+		move_still(ship_id);
+	}
+}
+// This file can safely be removed or replaced with your own code
+
+
+
+int main(int argc, char* argv[]){
+	char *c, inspiration_enabled_str[6];
+	unsigned char i, id, j, sca, scb, scc;
+	unsigned short s, ssa, ssb;
+	unsigned int sia;
+
+	// Firstly, we open a log file. If no log file is specified, it defaults to "MyBot.log"
+	if(argc != 2){
```

Comment:
This can be a 1 line statement.
`char *c = argc > 1 ? argv[1] : "MyBoy.log`

Reply:
I'm sorry, but how is what you suggested any clearer than what you just criticized me for with the complicated if statements that should be one-line functions? Perhaps it's just me, but I find ternary operators difficult to read. They're not any more difficult to read than an if statement with a tiny bit of complexity.